src/utils/notifications.ts

```diff
@@ -0,0 +1,144 @@
+import MagicBellClient, { Notification } from '@magicbell/core';
+import { query } from '~amplifyClient';
+import {
+  GetColonyContributorsNotificationDataDocument,
+  GetColonyContributorsNotificationDataQuery,
+  GetColonyContributorsNotificationDataQueryVariables,
+  NotificationsDataFragment,
+} from '~graphql';
+
+export enum NotificationType {
+  SimplePayment = 'SimplePayment',
+  Mention = 'Mention',
+}
+
+interface NotificationVariables {
+  title: string;
+  creator: string;
+  colonyAddress: string;
+  transactionHash: string;
+  notificationType: NotificationType;
+  amount?: string;
+  tokenAddress?: string;
+  mentions?: string[];
+}
+
+// Set up the notification client
+export const setupNotificationsClient = async (): Promise<void> => {
+  await MagicBellClient.createInstance({
+    apiKey: process.env.MAGICBELL_API_KEY as string,
+    apiSecret: process.env.MAGICBELL_API_SECRET as string,
+  });
+};
+
+// Check if notifications should be sent to a recipient from a colony.
+const shouldSendNotificationToRecipient = (
+  notificationsData: NotificationsDataFragment,
+  colonyAddress: string,
+): boolean => {
+  return (
+    !!notificationsData.magicbellUserId && // User has a magicbell user account made
+    !notificationsData.notificationsDisabled && // User has not disabled notifications app-wide
+    !notificationsData.mutedColonyAddresses.includes(colonyAddress) // User has not muted the colony that the notification is in
+  );
+};
+
+// Get all the recipients of a colony wide notification in the format Magicbell expects.
+const getRecipientsOfColonyWideNotification = async (
```

Comment:
Maybe it would make sense to group a bit the functionality here
So my proposal would be to first get all paginated members data
```
interface GetDataArgs extends Record<string, unknown> {
  nextToken?: string | null;
  colonyAddress: string;
}

interface GetDataResponse<TItem> {
  items: (TItem | null)[] | undefined,
  nextToken?: string | null;
}

type GetDataFn<T> = (args: GetDataArgs) => Promise<GetDataResponse<T> | null | undefined>

const getMembersData = async ({ colonyAddress, nextToken }: GetDataArgs): Promise<GetColonyContributorsNotificationDataQuery['getContributorsByColony']> => {
  const response = await query<
      GetColonyContributorsNotificationDataQuery,
      GetColonyContributorsNotificationDataQueryVariables
    >(GetColonyContributorsNotificationDataDocument, {
      colonyAddress,
      limit: 100,
      ...(nextToken ? { nextToken } : {}),
    })

    return response?.data?.getContributorsByColony
}

const getAllPages = async <T>(getData: GetDataFn<T>, params: GetDataArgs) => {
  const items = [];
  let nextToken = null;

  do {
    const data = await getData({ ...params, nextToken });
    nextToken = data?.nextToken;
    if (data?.items) {
      items.push(...data.items);
    }
  } while (nextToken);

  return items;
};

```

and then use it in this function as

```
const members = await getAllPages(getMembersData, { colonyAddress })

members.forEach(...)
```